### DISCUSSION TITLE
Interest in desktop app?

### DISCUSSION THREAD

**Discussion by Tzahi12345** [Tzahi12345/YoutubeDL-Material#409] [Show and tell]
There's been *some* code to support [Electron](https://www.electronjs.org/) for a while, but I've only recently put a serious attempt to try to get it working. In [one branch](https://github.com/Tzahi12345/YoutubeDL-Material/tree/electron-improvements), I'm playing around with the minor code changes necessary to get it spinning:

![image](https://user-images.githubusercontent.com/707413/127425629-a96c6e98-5040-4614-88aa-08aa5ef4b583.png)

It's still very early (thumbnails don't work for example), and it will require testing a new platform, but I could imagine it being a nice addition especially for those hesitant to self-host or install Docker. Let me know your thoughts!

EDIT:

Experimental build (Windows): https://github.com/Tzahi12345/YoutubeDL-Material/releases/download/v4.2/youtubedl-material-desktop-experimental.zip

**Comment by GlassedSilver** [comment]
If it optionally could integrate into a server hosted elsewhere and the desktop app would function as a front-end only, that could be interesting. Maybe even with conditional routing of requests, based on which network you're on. (local send requests to 192.168.xxx.xxx and when you're away from home, maybe connected through a VPN, send the requests to a different IP.

That's just an additional idea though, I see you want that application to function as a fully self-contained application. I think there's some usefulness to this, especially for users who are less tech savvy, however I'd advise to deliver this with a warning that unless you connect mongoDB to it scalability is severely hindered and on top of that, even with mongoDB connected, big databases are probably better handled by a later build.

Other than that, I think this is really interesting, especially if we consider that Microsoft decided to finally also let non-UWP apps make use of features like becoming a share sheet target.

Screw the Microsoft Windows App Store though. :P

**Reply by rockbo842** [reply]
I agree with this.
